src/db.py

```diff
@@ -199,8 +207,73 @@ def agent_finish_job(self, job: JobId) -> None:
         if new_agents <= 0:
             self.redis.hmset(job.key, {"status": "done"})
 
-    def register_active_agent(self, agent_id: str, ursadb_url: str) -> None:
-        self.redis.hset("agents", agent_id, ursadb_url)
+    def register_active_agent(
+        self,
+        agent_id: str,
+        ursadb_url: str,
+        plugins_spec: Dict[str, Dict[str, str]],
+        active_plugins: List[str],
+    ) -> None:
+        self.redis.hset(
+            "agents",
+            agent_id,
+            AgentSpecSchema(
+                ursadb_url=ursadb_url,
+                plugins_spec=plugins_spec,
+                active_plugins=active_plugins,
+            ).json(),
+        )
+
+    def get_active_agents(self) -> Dict[str, AgentSpecSchema]:
+        return {
+            name: AgentSpecSchema.parse_raw(spec)
+            for name, spec in self.redis.hgetall("agents").items()
+        }
+
+    def get_plugins_config(self) -> List[ConfigSchema]:
+        config_fields: Dict[str, Dict[str, str]] = {}
+        # Merge all config fields
+        for agent_spec in self.get_active_agents().values():
+            for name, fields in agent_spec.plugins_spec.items():
+                if name not in config_fields:
+                    config_fields[name] = {}
+                config_fields[name].update(fields)
+        # Transform fields into ConfigSchema
```

Comment:
Uhh, now I see what you meant when you said it's `a spaghetti` (because of our data format)